### PR TITLE
Update jet selection for 2017 (work with 94x jet pileup ID in 2017), lepton trigger SF,  Pisa DNN, fix dimuon mass to 125 GeV outside signal region in DNN evaluation

### PR DESCRIPTION
**Pull request by nanlu06**
Update jet selection for 2017 in [2.6,3.0] region. 

Reference: https://indico.cern.ch/event/863753/contributions/3640096/attachments/1945870/3228408/Raffaele_15_11_2019.pdf slide 11

### CODE REVIEW

**Comment by irenedutta23** [comment]
Hi Nan, I wanted to let you know that you probably haven't merged with Joosep's master branch .. this PR has an older version of run.sh and it is referring to an old singularity image which doesn't exist anymore. So it didn't run for me locally until I changed it.

**Comment by nanlu06** [comment]
Hi Irene, thank you very much for looking at this PR! Which singularity image are you changing to, and which is the old singularity image?

Thank you. @irenedutta23

**Comment by irenedutta23** [comment]
Hi Nan, perhaps you need to merge with Joosep's master branch. Then those changes should be reflected automatically. I did that and it worked for me.

**Comment by nanlu06** [comment]
Thanks Irene for your reply. I started with Joosep's master branch when I created this PR so I am confused which old singularity image you are referring to. @irenedutta23

**Comment by irenedutta23** [comment]
Hi Nan, ok yes, I see that now it has the right singularity image. Perhaps I had downloaded your master branch first which wasn't merged to the latest master branch from Joosep. But the build still fails. 
https://travis-ci.com/jpata/hepaccelerate-cms/builds/144266158
Perhaps @jpata knows why this build fails?

**Comment by jpata** [comment]
The automatic test fails before our code even starts to run, something to do with tensorflow installation on the testing machine in the cloud. I think it's unrelated to this PR and I will need to look into it separately. For the moment, we will need to do our own tests on T2 machines for the PR.

**Comment by irenedutta23** [comment]
Hi Nan, 
Since you implemented fixing the Higgs mass to 125. outside the H peak region. could you also fix it here in this line (if it would convolute the main agenda of this PR, I can open a separate small PR too):
https://github.com/nanlu06/hepaccelerate-cms/blob/my_new_fix/tests/hmm/hmumu_utils.py#L2524

Thank you very much! @nanlu06

**Comment by nanlu06** [comment]
Hi Irene! Thank you for noticing this.

So I need to keep "Higgs_mass" as it is because there are some other variables calculated using the actual dimuon mass.

I do see "Higgs_mass" in your DNN vars: https://github.com/nanlu06/hepaccelerate-cms/blob/87ffa10389539122b95910ddb58933bfcfeb7bdc/tests/hmm/pars.py#L606 So I would propose to change "Higgs_mass" to "Higgs_m" in your DNN vars. I will do that.

**Comment by irenedutta23** [comment]
Hi @nanlu06 
I produced some validation plots for the Pisa DNN for this PR.
https://cernbox.cern.ch/index.php/s/oOaWKzFgB9Gt4vR

The JEC band seems too high .. But maybe it's expected. There seems to be some issue about how the Higgs_m or Higgs_mass is getting used for the calculation of our DNN, which still needs to be fixed.Also the distribution of the dEta_jj variable is skewed here too, as we saw previously. Both these issues  need to be fixed. I'm looking into it. Other things seem fine to me.
Thanks!

**Comment by nanlu06** [comment]
@irenedutta23 changes implemented as you suggested.

File: tests/hmm/hmumu_utils.py [hunk 1]
```diff
@@ -2874,16 +2899,27 @@ def multiply_all(weight_list):
         ret *= w
     return ret
 
-def compute_lepton_sf(leading_muon, subleading_muon, lepsf_iso, lepsf_id, lepsf_trig, use_cuda, dataset_era, NUMPY_LIB, debug):
+def multiply_all_trig(weight_list):
```
Comment (jpata) [inline tests/hmm/hmumu_utils.py]: maybe we can put a comment here that this computes (as I understand) `1 - (1 - w1)(1 - w2)(1 - w3)...`

**Comment by irenedutta23** [comment]
Hi Nan, I was checking the datacards and histograms and I still see the DNN prediction values being incorrect with the new commit. I'm sure it has to do with the 'Higgs_m' variable because if I just use 'Higgs_mass=fixm', the dnn prediction is right. Something needs to be still changed in the plotting.py file or the pars.py file.

File: tests/hmm/run.sh [hunk 2]
```diff
@@ -20,7 +20,11 @@ export KERAS_BACKEND=tensorflow
 
 #This is the location of the input NanoAOD and generally does not need to be changed
 export INPUTDATAPATH=/storage/group/allcit
+
+#nanoAODv5
 export CACHEPATH=/storage/user/jpata/hmm/skim_merged
+#nanoAODv6
+#export CACHEPATH=/storage/user/nlu/hmm/skim_merged
```
Comment (jpata) [inline tests/hmm/run.sh]: I think we can change the default to the private nanoAODv6 now and take this as the new baseline?

**Comment by nanlu06** [comment]
@jpata @irenedutta23 Thanks a lot for the checks.

Irene agreed to merge this one and have another PR to fix how the Dimuon mass is fixed.

please note the jet selection change in this PR applies to this:
datasets_NanoAODv6_2017_nanlu_privateproduction.yml
 
because nanoAODv5 does not have 9X (2017)[1] pileup jet ID. 

[1: https://twiki.cern.ch/twiki/bin/view/CMS/PileupJetID#9X_2017_and_10X_2018_recipes]

**Pull request merged**